test/e2e/metrics_test.go

```diff
@@ -0,0 +1,123 @@
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	test "github.com/openshift/cluster-kube-apiserver-operator/test/library"
+)
+
+func TestMetricsRegistration(t *testing.T) {
+	kubeConfig, err := test.NewClientConfigForTest()
+	require.NoError(t, err)
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	restConfig, err := test.NewClientConfigForTest()
+	require.NoError(t, err)
+	restConfig = setRESTConfigDefaults(restConfig)
+	restClient, err := rest.RESTClientFor(restConfig)
+	require.NoError(t, err)
+
+	// get list of operator pods
+	pods, err := kubeClient.CoreV1().Pods(operatorclient.OperatorNamespace).List(metav1.ListOptions{})
+	require.NoError(t, err)
+
+	// we just care about the one we expect
+	require.GreaterOrEqual(t, len(pods.Items), 1)
+	pod := pods.Items[0]
+
+	metrics := getMetricsForPod(t, pod, kubeConfig, restClient)
+	t.Logf("Retrieved %d metrics.", len(metrics))
+	if len(metrics) == 0 {
+		t.Fatal("No metrics retrieved.")
+	}
+}
+
+func getMetricsForPod(t *testing.T, pod v1.Pod, kubeConfig *rest.Config, restClient *rest.RESTClient) map[string]string {
+	require.Equal(t, v1.PodRunning, pod.Status.Phase)
+	transport, upgrader, err := spdy.RoundTripperFor(kubeConfig)
+	require.NoError(t, err)
+	request := restClient.Post().Resource("pods").Namespace(pod.Namespace).Name(pod.Name).SubResource("portforward")
+	dialer := spdy.NewDialer(upgrader, &http.Client{Transport: transport}, http.MethodPost, request.URL())
+	ports := []string{"37587:8443"}
+	stop := make(chan struct{}, 1)
+	ready := make(chan struct{}, 1)
+	data := bytes.NewBuffer(nil)
+	portForwarder, err := portforward.New(dialer, ports, stop, ready, bytes.NewBuffer(nil), ioutil.Discard)
+	require.NoError(t, err)
+	go func() {
+		if stop != nil {
+			defer close(stop)
+		}
+		t.Log("Waiting for port forwarder to be ready...")
+		<-ready
+		t.Log("Ready.")
+		config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			clientcmd.NewDefaultClientConfigLoadingRules(),
+			&clientcmd.ConfigOverrides{
+				ClusterInfo: api.Cluster{
+					InsecureSkipTLSVerify: true,
+					Server:                "https://localhost:37587",
+				},
+			},
+		).ClientConfig()
+		config = setRESTConfigDefaults(config)
+		require.NoError(t, err)
+		restClient, err := rest.RESTClientFor(config)
+		require.NoError(t, err)
+		t.Log("Retrieving metrics...")
+		stream, err := restClient.Get().RequestURI("/metrics").Stream()
+		require.NoError(t, err)
+		defer stream.Close()
+		_, err = io.Copy(data, stream)
+	}()
+	t.Log("Starting port forwarder...")
+	err = portForwarder.ForwardPorts()
+	require.NoError(t, err)
+	<-stop
+	metrics := map[string]string{}
+	scanner := bufio.NewScanner(data)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+		entry := strings.SplitN(line, " ", 2)
+		metrics[entry[0]] = entry[1]
+	}
+	require.NoError(t, scanner.Err())
+	return metrics
```

Comment:
make the test a bit easier to read with empty lines and high-level oneliner comments.

Reply:
Done.